Makefile

```diff
@@ -96,6 +121,31 @@ else
 	@$(call check_program,git)
 endif
 
+
+install-image: $(IMAGE) $(DEFAULT_RUNTIME_CONFIG) $(DESTSYSCONFIG)
+	install -D --owner root --group root --mode 0644 "$(IMAGE)" "$(IMAGE_DEST)"
+	@echo Adding $(IMAGE_DEST) to configuration.
+	@sed -i.bak -e 's!^\(image = ".*"\)!# \1\nimage = "$(IMAGE_DEST)"!g' \
+		 $(DESTSYSCONFIG)
+
+get-machine-type: $(DESTSYSCONFIG)
+	@echo Checking machine type ...
+	$(eval MACHINE_TYPE=$(shell grep -P "^machine_type\s*=\s*" $(DESTSYSCONFIG) | grep -oP '"\K.*(?=")') )
+	@echo Machine type is $(MACHINE_TYPE)
+
+install-kernel: $(VMLINUZ) $(VMLINUX) $(DESTSYSCONFIG) get-machine-type
+	install -D --owner root --group root --mode 0644 $(VMLINUX) $(VMLINUX_DEST)
+	install -D --owner root --group root --mode 0644 $(VMLINUZ) $(VMLINUZ_DEST)
+	@echo Configured to use machine type \"$(MACHINE_TYPE)\" using kernel $(KERNEL_PATH)
+	@sed -i -e 's!^\(kernel = ".*"\)!# \1\nkernel= "$(KERNEL_PATH)"!g' \
+		$(DESTSYSCONFIG)
+
+$(DESTSYSCONFIG): $(DESTCONFIG)
+	mkdir -p $(DESTSYSCONFDIR)
+	@echo Generating $(DESTSYSCONFIG) from $(DESTCONFIG)
+	@echo "# XXX: WARNING: this config was generated by osbuilder on $$(date) from $(DESTCONFIG)" > $(DESTSYSCONFIG)
```

Comment:
Nice. We will end up with two warnings of course (as `/usr/share/defaults/clear-containers/configuration.toml` already contains such a message as it too is auto-generated), but since we have the timestamp in there and the osbuilder Warning comes first, it should be clear enough ;)